doc/port_link_training/port-link-training-design.md

```diff
@@ -0,0 +1,336 @@
+  Ethernet8        off         off             -              -    down       up
+ Ethernet16         on        auto          none        trained      up       up
+ Ethernet24        off           -             -              -    down       up
+ Ethernet32        off           -             -              -    down       up
+```
+
+
+#### Config DB Enhancements  
+
+A new field **link_training** will be added to **PORT** table:
+
+	; Defines information for port configuration
+	key                     = PORT|port_name     ; configuration of the port
+	; field                 = value
+	...
+	link_training           = STRING             ; link training configuration
+
+Valid value of the new fields are described below:
+
+- link_training:  
+String value, the administratively specified port link training configuration.
+When "link_training" is not specified, the port link training should be disabled  
+
+ | Mode |     Description                                                 |
+ |:----:|:---------------------------------------------------------------:|
+ | auto | Enable link-training if applicable to the transceiver attached  |
+ | on   | Enable link-training regardless of the transceiver type         |
+ | off  | Disable link-training                                           |
+
+For deployment considerations, users are encouraged to use only **auto** and **off** for the **link_training**, while the **on** mode is for test purpose, it should only be activated when the transceiver is not correctly identified by the pmon#xcvrd, and forcing link-training enabled is intended.
+
+#### Application DB Enhancements
+
+The following fields will be introduced into **PORT_TABLE** table:
+
+	; Defines information for port configuration
+	key                     = PORT_TABLE:port_name ; configuration of the port
+	; field                 = value
+	...
+	link_training           = STRING               ; operational link training config
+	link_training_status    = STRING               ; operational link training status
+	link_training_failure   = STRING               ; operational link training failure status
+	link_training_rxstatus  = STRING               ; operational link training rx status
+	xcvr_capabilities       = STRING               ; transceiver capabilities
+
+- link_training:  
+String value, the admin port link training config.
+- link_training_status:  
+String value, the operational port link training status. "on" indicates port link training is enabled, otherwise disabled.
+- link_training_failure:  
+String value, the operational port link training failure status. "none" indicates no error is detected, otherwise any of the link training failure status defined in SAI. For example: "none", "lock", "snr" and "timeout".
+- link_training_rxstatus:  
+String value, the operational port link training rx status. any of the link training rx status defined in SAI. For example: "trained", "not-trained".
+- xcvr_capabilities:
```

Comment:
What are all the list of values that can be in xcr_capabilities apart from LT. Please detail it in the HLD.